### PR TITLE
Share Structure Functionality Updates and Improvements

### PR DESCRIPTION
**Pull request by cameron-freshworks**
*Issue #:* /bcgov/entity#5202

*Description of changes:*

* Updated and improved functionality of Share Structure correcting as per updated requirements.

* Added series edit form to show the edit form in the proper location when correcting either a Class or Series.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

File: src/components/ShareStructure/ShareStructure.vue [hunk 1]
```diff
@@ -149,11 +163,11 @@
             </div>
           </td>
         </tr>
-        <tr v-if="showEditShareStructureForm[row.index]">
+        <tr v-if="showClassEditForm[row.index]">
           <td colspan="6">
             <div class="edit-share-structure-container">
               <edit-share-structure
-                v-show="showEditShareStructureForm"
+                v-show="showClassEditForm"
```
Comment (severinbeauvais) [inline src/components/ShareStructure/ShareStructure.vue:170]: Looking at lines 166 and 170, something doesn't seem right. Is `showClassEditForm` an array or a boolean?

**Review by severinbeauvais — COMMENTED** [review]
Updated unit tests?

**Review by vysakh-menon-aot — APPROVED** [review]

File: src/components/ShareStructure/ShareStructure.vue (hunk 1 above)
Comment (cameron-freshworks) [reply src/components/ShareStructure/ShareStructure.vue:170]: It is an Array of booleans which are false by default. The index of the array is the index of the row of either the series share or class. The line you highlighted above though, was a redundant check and been removed :)

**Comment by cameron-freshworks** [comment]
> Updated unit tests?

Will add a few more! Next commit

**Comment by cameron-freshworks** [comment]
> Updated unit tests?

I have unit tests that shuffle the indexes around testing the move up / down functionality. It really threw me off!
I love unit tests, they live in a vacuum and never lie 😆  Updated*

**Review by severinbeauvais — APPROVED** [review]

**Pull request merged**